### ISSUE TITLE
[🐛 BUG] Improve and fix stylekit to make UI elements more consistent

### ISSUE THREAD

**Issue by FlorianJacta** [Avaiga/taipy#2305]
### What went wrong? 🤔

Here is a list of UI components. The stylekit should provide a consistent set of UI elements. Some viz elements are higher than others.

![image](https://github.com/user-attachments/assets/f8ec9230-c224-4eb9-8d08-11f2d85d734a)

### Expected Behavior

Some elements should be fixed, and others may be improved. I don't have any specific recommendations for improvement.

### Steps to Reproduce Issue

Run this code.

```python
from taipy.gui import Gui, notify
import taipy.gui.builder as tgb


with tgb.Page() as page:
    with tgb.layout(columns="1 1 1 1 1"):
        tgb.selector(dropdown=True)
        tgb.selector(dropdown=True, filter=True)
        tgb.selector(
            dropdown=True,
            label="City",
        )
        tgb.selector(dropdown=True, label="City", filter=True)
        tgb.input("Some Text")
        tgb.input("Some Text", label="Write some text")
        tgb.number(10)
        tgb.number(10, label="Write a number")
        tgb.date("2021-01-01")
        tgb.date("2021-01-01", label="Pick a date")
        tgb.date("2021-01-01", with_time=True)
        tgb.date_range(["2021-01-01", "2022-01-01"])
        tgb.date_range(
            ["2021-01-01", "2022-01-01"], label_start="Start", label_end="End"
        )
        tgb.slider(10)
        tgb.slider(10, label="Pick a number")  # No label
        tgb.toggle(lov=["Yes", "No"])
        tgb.toggle(lov=["Yes", "No"], label="Yes or No")  # Misplaced label
        tgb.toggle(True)
        tgb.toggle(True, label="Yes or No")
        tgb.button("Click me")
        tgb.button("Click me full and plain", class_name="fullwidth plain")
        tgb.file_download()
        tgb.file_selector()
        tgb.file_selector(label="Pick a file")
        tgb.status("Ready")


Gui(page).run()
```

## Other example

https://github.com/Avaiga/taipy/issues/2262

Visual elements should have the similar position and layouts in the default stylekit. For example, if I place a selector and a button side by side, they should be aligned:

```python
from taipy.gui import Gui
import taipy.gui.builder as tgb


with tgb.Page() as page:
    with tgb.layout(columns="1 1 10"):
        with tgb.part():
            tgb.button("Click me")
        with tgb.part():
            tgb.selector("{selected}", lov="Furniture;Desk", dropdown=True)

Gui(page=page).run()
```

Here, they are misaligned:
![image](https://github.com/user-attachments/assets/64b3db15-3425-4d4b-8723-e85e1e6a5e1c)

This issue makes layouts a CSS nightmare in any application:
![image](https://github.com/user-attachments/assets/c5ddc14e-c27b-40f0-86cb-16b459ec6f41)

### Browsers

Chrome

### OS

Windows

### Version of Taipy

4.0.1

### Acceptance Criteria

- [ ] A unit test reproducing the bug is added.
- [ ] Any new code is covered by a unit tested.
- [ ] Check code coverage is at least 90%.
- [ ] The bug reporter validated the fix.
- [ ] Related issue(s) in taipy-doc are created for documentation and Release Notes are updated.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

**Comment by FredLL-Avaiga** [comment]
![Image](https://github.com/user-attachments/assets/93338623-dcd3-4e46-927e-0f71f86cfa8c)

**Comment by FlorianJacta** [comment]
Do you have this example too? 

```python
from taipy.gui import Gui
import taipy.gui.builder as tgb


with tgb.Page() as page:
    with tgb.layout(columns="1 1 10"):
        with tgb.part():
            tgb.button("Click me")
        with tgb.part():
            tgb.selector("{selected}", lov="Furniture;Desk", dropdown=True)

Gui(page=page).run()
```

**Comment by FredLL-Avaiga** [comment]
![image](https://github.com/user-attachments/assets/e42c4c84-d858-4339-ae85-47ebadfa7688)
When I change taipy-selector's margin to 0 in selector.css (I'll side load this in my current PR)

**Comment by FlorianJacta** [comment]
There is still a slight issue of alignment here
![image](https://github.com/user-attachments/assets/6959cfa1-9553-40dc-b077-d48b7903dd5e)

**Comment by FredLL-Avaiga** [comment]
Some code?

**Comment by FlorianJacta** [comment]
It is the one from the issue. The top right of the screen

**Comment by AlexandreSajus** [comment]
Fixed in 4.0.2.dev2. Beautiful!
